qdev_wrappers/parameters.py

```diff
@@ -0,0 +1,83 @@
+from typing import Callable
+
+from qcodes.instrument.parameter import Parameter, MultiParameter
+from qcodes.utils import validators
+
+
+class GroupSetParameter(Parameter):
+    """
+    Group a number of parameters to be set simultaneaously.
+    >>> p = GroupSetParameter('bound_channels', qdac.ch01.v, qdac.ch02.v)
+    >>> p(1)
+    here qdac.ch01.v and qdac.ch02.v will both set to 1V
+
+    Beware that setting a stepping does apply to this parameter
+    and is not inherited to the group of parameters.
+
+    max_val_age not supported
+    """
+    def __init__(self,
+                 name: str,
+                 *parameters,
+                 pre_set: Callable=None,
+                 post_set: Callable=None,
+                 **base_kwargs) -> None:
+        # this class has a different footprint that the Parameter base class
+        # because here the args must be given as kwargs.
+        # this is fine for the Instrument.add_parameter method as it uses
+        # kwargs.
+
+        # type checking for compatibility
+        if not all(isinstance(parameter, Parameter)
+                   for parameter in parameters):
+            raise RuntimeError("All supplied parameters must be of type Parameter")
+        # include multiparameter and arrayparameter case at some point
+        if any(isinstance(parameter, MultiParameter)
```

Comment:
Can we even reach this code path. AFAIK MultiParameter is not a subclass of Parameter (but of baseparameter)

Reply:
Well spotted, this has to be _BaseParameter of course